x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -595,52 +668,13 @@ private HitExtractor createExtractor(FieldExtraction ref) {
             this.schema = Rows.schema(output);
         }
 
-        // TODO: need to handle rejections plus check failures (shard size, etc...)
         @Override
         public void onResponse(final SearchResponse response) {
-            try {
-                ShardSearchFailure[] failure = response.getShardFailures();
-                if (CollectionUtils.isEmpty(failure) == false) {
-                    cleanup(response, new SqlIllegalArgumentException(failure[0].reason(), failure[0].getCause()));
-                } else {
-                    handleResponse(response, ActionListener.wrap(delegate::onResponse, e -> cleanup(response, e)));
-                }
-            } catch (Exception ex) {
-                cleanup(response, ex);
-            }
+            handleResponse(response, delegate);
```

Comment:
The handling of search request failures and subsequent cleaning up of PIT has been moved to `Querier.searchWithPointInTime` and is currently only applied to SearchHit queries. The other kind of queries do not need the cleanup logic and `ShardSearchFailure`s should no longer be a concern because search request use `allow_partial_search_results = false`.

Reply:
Cleanup is still done like before but now according to the search context that's in use (closing PIT for search hit queries or doing nothing for all other queries atm). Leaving it here (in addition to the cleanup in `Querier.searchWithPointInTime`) would only duplicate some of the cleanup logic.

An alternative would be to move the cleanup logic from `Querier.searchWithPointInTime` here. This would separate opening PIT from closing it and also tear apart the clean up logic. Before, clean up only needed to be performed when an error happened while processing the search response, now with PIT we also need to clean up when something happens between getting the PIT and getting the first search response (e.g. the search request fails).

Reply:
What case are you thinking of @bpintea ? `allow_partial_search_results = false` should ensure that this is never the case.